### PR TITLE
+build Add sort imports support.

### PR DESCRIPTION
**Pull request by He-Pin**
Sort and group imports for better code style and navigation.
Run `sortImports` in sbt shell to sort imports.
Currently there an issue about inline import,refs:https://github.com/NeQuissimus/sort-imports/issues/26
~~will update to liancheng/scalafix-organize-imports~~
The sort-imports is faster than the scalafix-organize-imports.

### CODE REVIEW

File: .scalafix.conf [hunk 1]
```diff
@@ -20,3 +21,12 @@ ignored-packages = [
...
   "jdoc"
 ]
+
+//sort imports, see https://github.com/NeQuissimus/sort-imports
+SortImports.blocks = [
+  "java.",
+  "scala.",
+  "*",
+  "com.sun."
+  "akka."
```
Comment (hepin1989) [inline .scalafix.conf]: With this kind of order

**Comment by raboof** [comment]
I'm on the fence, but I think I kindof like this.

We would have to disable it for documentation, however, since there we have the imports carefully manually ordered to make sure only the imports that are relevant for the given example are included in that example, see https://github.com/akka/akka/pull/28780/files#diff-19469208afa395ca09490d26db477b4cR14 .

**Comment by hepin1989** [comment]
> https://github.com/akka/akka/pull/28780/files#diff-19469208afa395ca09490d26db477b4cR14

It's disabled by for the `docs`

**Comment by raboof** [comment]
Ah, you're right - I just stumbled upon a snippet that is included from the tests, for example in akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala

We still have a bunch of these, though perhaps we should move them to akka-docs?

**Comment by hepin1989** [comment]
> Ah, you're right - I just stumbled upon a snippet that is included from the tests, for example in akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala
> 
> We still have a bunch of these, though perhaps we should move them to akka-docs?

Yes, there are tests been used in paradox, should I exclude them or move them into `akka-docs` I have no idea for now.

**Comment by patriknw** [comment]
> We still have a bunch of these, though perhaps we should move them to akka-docs?

I don't think we should move all doc snippets to akka-docs, because it's better developer experience for **us** to have it in respective project. In recent work we have such in tests but in a separate docs/jdocs package. Can we exclude it based on such package prefix, and then maybe also exclude a few more specific test where they are embedded in normal tests, such as the mentioned SourceSpec.scala

**Review by patriknw — COMMENTED** [review]
Looking good. I like that it's not messing with the import groups `a.b.{c, d, e}`. Should be possible to have a IntelliJ configuration that is compatible with this order.

**Comment by hepin1989** [comment]
`Can we exclude it based on such package prefix,` I think that should need more work, I can take a try with that.

File: .scalafix.conf [hunk 2]
```diff
@@ -12,11 +13,22 @@ ignored-files = [
   "FlowPrependSpec.scala",
   "FlowZipSpec.scala",
   "FlowZipWithSpec.scala",
-  "FlowZipWithIndexSpec.scala"
+  "FlowZipWithIndexSpec.scala",
+  "SourceSpec.scala",
+  "StatsSampleSpec.scala"
```
Comment (hepin1989) [inline .scalafix.conf]: Can do exclude here, @patriknw

File: .scalafix.conf [hunk 3]
```diff
@@ -12,11 +13,23 @@ ignored-files = [
...
+  "SourceSpec.scala",
+  "StatsSampleSpec.scala",
+  "ActorFlowSpec.scala"
 ]
 
 //ignored packages
 ignored-packages = [
   "doc",
   "jdoc"
 ]
```
Comment (patriknw) [inline .scalafix.conf]: So these packages will be ignored for the sorting of the imports?

I think we haven't been fully consistent with the naming of those packages. Please also add `docs` and `jdocs`.

Comment (hepin1989) [reply .scalafix.conf]: done，`doc` 、`docs` and `jdoc` are both added.

**Comment by ennru** [comment]
Replaces https://github.com/akka/akka/pull/28761

**Review by ennru — APPROVED** [review]
I like it as long as it doesn't become another build step...
LGTM.

**Comment by hepin1989** [comment]
> I like it as long as it doesn't become another build step...
> LGTM.

It isn't. I think there is no need to make id a build step, we can run the format occasionally.

**Comment by hepin1989** [comment]
Will update to liancheng/scalafix-organize-imports

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
thanks @hepin1989

**Pull request merged**